Indicators/ImpliedVolatility.cs

```diff
@@ -175,32 +195,205 @@ public class ImpliedVolatility : OptionIndicatorBase
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="option">The option to be tracked</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
         /// <param name="riskFreeRate">Risk-free rate, as a constant</param>
         /// <param name="dividendYield">Dividend yield, as a constant</param>
+        /// <param name="optionModel">The option pricing model used to estimate IV</param>
         /// <param name="period">The lookback period of historical volatility</param>
+        public ImpliedVolatility(string name, Symbol option, Symbol mirrorOption, decimal riskFreeRate = 0.05m, decimal dividendYield = 0.0m,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, int period = 252)
+            : this(name, option, mirrorOption, new ConstantRiskFreeRateInterestRateModel(riskFreeRate), new ConstantDividendYieldModel(dividendYield), optionModel, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the ImpliedVolatility class
+        /// </summary>
+        /// <param name="option">The option to be tracked</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
+        /// <param name="riskFreeRate">Risk-free rate, as a constant</param>
+        /// <param name="dividendYield">Dividend yield, as a constant</param>
         /// <param name="optionModel">The option pricing model used to estimate IV</param>
-        public ImpliedVolatility(Symbol option, decimal riskFreeRate = 0.05m, decimal dividendYield = 0.0m, int period = 252,
-            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes)
-            : this($"IV({option.Value},{period},{riskFreeRate},{dividendYield},{optionModel})", option, riskFreeRate, dividendYield, period, optionModel)
+        /// <param name="period">The lookback period of historical volatility</param>
+        public ImpliedVolatility(Symbol option, Symbol mirrorOption, decimal riskFreeRate = 0.05m, decimal dividendYield = 0.0m,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, int period = 252)
+            : this($"IV({option.Value},{mirrorOption.Value},{riskFreeRate},{dividendYield},{optionModel},{period})", option, mirrorOption, riskFreeRate, dividendYield, optionModel, period)
         {
         }
 
+        private bool _isReady => Price.Current.Time == UnderlyingPrice.Current.Time && Price.IsReady && UnderlyingPrice.IsReady;
+
         /// <summary>
         /// Gets a flag indicating when this indicator is ready and fully initialized
         /// </summary>
-        public override bool IsReady => HistoricalVolatility.Samples >= 2 && Price.Current.Time == UnderlyingPrice.Current.Time;
+        public override bool IsReady => UseMirrorContract ? _isReady && Price.Current.Time == _oppositePrice.Current.Time && _oppositePrice.IsReady : _isReady;
```

Comment:
I think this will cause an unexpected behavior where IsReady would toggle from true to false, depending on the time of each sub indicator being used? It seems once the condition is meet once it should stick as IsReady true until reset?

Reply:
We intend to make it this way, since we need all 3 underlying indicators having their timestamp allign to calculate the fair price and greeks